core/src/main/java/com/linecorp/armeria/client/HttpRequestSubscriber.java

```diff
@@ -298,7 +322,7 @@ private void cancelSubscription() {
         subscription.cancel();
     }
 
-    private void failAndRespond(Throwable cause) {
+    private void failAndWriteResetIfActive(Throwable cause) {
```

Comment:
I think `failAndReset()` is enough. Whether a channel is active or not doesn't matter much because the RST_STREAM frame would not be sent even if we didn't have `if (ch.isActive())`.

Reply:
Thanks fixed. 😄